File: CastIt.Domain/Dtos/Responses/FileItemResponseDto.cs
using System;
using System.Collections.Generic;

namespace CastIt.Domain.Dtos.Responses
{
    public class FileItemResponseDto
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double TotalSeconds { get; set; }
        public string Path { get; set; }
        public int Position { get; set; }
        public double PlayedPercentage { get; set; }
        public long PlayListId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public bool Loop { get; set; }

        public bool IsBeingPlayed { get; set; }
        public bool IsLocalFile { get; set; }
        public bool IsUrlFile { get; set; }
        public double PlayedSeconds { get; set; }
        public bool CanStartPlayingFromCurrentPercentage { get; set; }
        public bool WasPlayed { get; set; }
        public bool IsCached { get; set; }

        public bool Exists { get; set; }
        public string Filename { get; set; }
        public string Size { get; set; }
        public string Extension { get; set; }

        public string SubTitle { get; set; }
        public string Resolution { get; set; }
        public string Duration { get; set; }
        public string PlayedTime { get; set; }
        public string TotalDuration { get; set; }
        public string FullTotalDuration { get; set; }
        public string ThumbnailUrl { get; set; }

        public List<FileItemOptionsResponseDto> CurrentFileVideos { get; set; }
            = new List<FileItemOptionsResponseDto>();
        public List<FileItemOptionsResponseDto> CurrentFileAudios { get; set; }
            = new List<FileItemOptionsResponseDto>();
        public List<FileItemOptionsResponseDto> CurrentFileSubTitles { get; set; }
            = new List<FileItemOptionsResponseDto>();
        public List<FileItemOptionsResponseDto> CurrentFileQualities { get; set; }
            = new List<FileItemOptionsResponseDto>();
        public int CurrentFileVideoStreamIndex { get; set; }
        public int CurrentFileAudioStreamIndex { get; set; }
        public int CurrentFileSubTitleStreamIndex { get; set; }
        public int CurrentFileQuality { get; set; }
    }
}

File: CastIt.Client/lib/domain/models/dtos/responses/file_item_response_dto.dart
import 'package:castit/domain/models/models.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'file_item_response_dto.freezed.dart';
part 'file_item_response_dto.g.dart';

@freezed
sealed class FileItemResponseDto with _$FileItemResponseDto {
  List<FileItemOptionsResponseDto> get streams =>
      currentFileVideos + currentFileAudios + currentFileSubTitles + currentFileQualities;

  factory FileItemResponseDto({
    required int id,
    String? name,
    String? description,
    required double totalSeconds,
    required String path,
    required int position,
    required double playedPercentage,
    required int playListId,
    required DateTime createdAt,
    DateTime? updatedAt,
    DateTime? lastPlayedDate,
    required bool loop,
    required bool isBeingPlayed,
    required bool isLocalFile,
    required bool isUrlFile,
    required double playedSeconds,
    required bool canStartPlayingFromCurrentPercentage,
    required bool wasPlayed,
    required bool isCached,
    required bool exists,
    required String filename,
    required String size,
    required String extension,
    required String subTitle,
    required String duration,
    required String playedTime,
    required String totalDuration,
    required String fullTotalDuration,
    required List<FileItemOptionsResponseDto> currentFileVideos,
    required List<FileItemOptionsResponseDto> currentFileAudios,
    required List<FileItemOptionsResponseDto> currentFileSubTitles,
    required List<FileItemOptionsResponseDto> currentFileQualities,
    required int currentFileVideoStreamIndex,
    required int currentFileAudioStreamIndex,
    required int currentFileSubTitleStreamIndex,
    required int currentFileQuality,
    String? thumbnailUrl,
  }) = _FileItemResponseDto;

  FileItemResponseDto._();

  factory FileItemResponseDto.fromJson(Map<String, dynamic> json) => _$FileItemResponseDtoFromJson(json);
}
